### ISSUE TITLE
NullReferenceException caused by Robotic Parts

### ISSUE THREAD

**Issue by Lucaspec72** [KSPModdingLibs/KSPCommunityFixes#25]
Not 100% sure this isn't caused by another mod, but my console is flooded with NullReferenceExceptions and the log points to KSPCommunityFixes.

```
[EXC 10:33:34.877] NullReferenceException: Object reference not set to an instance of an object
KSPCommunityFixes.BugFixes.RoboticsDrift+ServoInfo.PristineCoordsUpdate ()
KSPCommunityFixes.BugFixes.RoboticsDrift.BaseServo_RecurseCoordUpdate_Prefix (Expansions.Serenity.BaseServo __instance, Part p, UnityEngine.ConfigurableJoint ___servoJoint, UnityEngine.GameObject ___movingPartObject)
Expansions.Serenity.BaseServo.Expansions.Serenity.BaseServo.RecurseCoordUpdate_Patch1(Expansions.Serenity.BaseServo,Part,Part)
Expansions.Serenity.BaseServo.ApplyCoordsUpdate()
Expansions.Serenity.BaseServo.Update()
UnityEngine.DebugLogHandler:LogException(Exception, Object)
ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```

_Originally posted by @Lucaspec72 in https://github.com/KSPModdingLibs/KSPCommunityFixes/issues/13#issuecomment-1077437663_

**Comment by gotmachine** [comment]
Can you post your full KSP.log and save file (or better, use [KSPBugReport ](https://github.com/KSPModdingLibs/KSPBugReport), reproduce the issue and upload a bug report) ?
Also, can you expand a bit on under what conditions this happens ?
Does it happen immediately after launching the vessel or is it a vessel that was already in flight in an existing save ?
At least can you post a screenshot of your vessel showing which robotic parts you're using and under which configuration ?

There isn't much I can do if you don't provide me a way to reproduce that issue.

**Comment by Lucaspec72** [comment]
Didn't do much testing, but it seems to happen whenever any robotic part is on a vessel. i noticed the bug when trying to go back to the KSC from flight after launching a vessel (all save/exit options where broken). I removed KSPCF temporarly to fix this, i'll reinstall it and do a bug report in the evening.
As of which parts i'm using, it's only a piston, a couple of servomotors, a hinge and i forgot the name of the last one.
![image](https://user-images.githubusercontent.com/48445530/159902045-e1823921-f4ff-4210-a59b-38f906e7ebd0.png)

**Comment by gotmachine** [comment]
Well, if you manage to reproduce it with a stock craft using no modded parts, please upload the craft file. That would help me a lot.

I have done a fair bit of testing with a a good dozen of complex robotic crafts grabbed on KerbalX and that issue never happened, so there must be something quite specific either in your KSP install or craft setup that is triggering this.

**Comment by gotmachine** [comment]
I will close this as I I can't reproduce it without more instructions.
Moreover, it is likely that this exception was caused by bug that was fixed in release 1.14.0
